addons/project/views/project_portal_templates.xml

```diff
@@ -137,15 +137,19 @@
             </t>
             <div t-if="tasks" class="panel panel-default">
                 <div class="table-responsive"><table class="table table-hover o_portal_my_doc_table">
-                    <thead>
-                        <tr class="active">
-                            <th class="col-md-9">Task</th>
-                            <th class="col-md-2">Stage</th>
-                            <th>Ref</th>
-                        </tr>
-                    </thead>
-                    <tbody>
-                        <t t-foreach="tasks" t-as="task">
+                    <t t-set="project_id" t-value="''"/>
+                    <t t-foreach="tasks" t-as="task">
+                        <t t-if="project_id != task.project_id">
+                            <t t-set="project_id" t-value="task.project_id"/>
+                            <thead>
```

Comment:
If I am not wrong, in html you should have only 1 thead by table ... but we can have th instead of td if I am not wrong.